test/functional/job_configs.py

```diff
@@ -24,21 +24,16 @@ def __init__(self, config, expected_to_fail, expected_num_subjobs, expected_num_
 # This is a very basic job where each atom just creates a simple text file.
 BASIC_JOB = FunctionalTestJobConfig(
     config={
-        'posix': """
-BasicJob:
-    commands:
-        - echo $TOKEN > $ARTIFACT_DIR/result.txt
-    atomizers:
-        - TOKEN: seq 0 4 | xargs -I {} echo "This is atom {}"
-
-""",
-        'nt': """
-BasicJob:
-    commands:
-        - echo !TOKEN!> !ARTIFACT_DIR!\\result.txt
-    atomizers:
-        - TOKEN: FOR /l %n in (0,1,4) DO @echo This is atom %n
-""",
+        'posix':
+            {
+                'commands': ['echo $TOKEN > $ARTIFACT_DIR/result.txt'],
+                'atomizers': [{'TOKEN': 'seq 0 4 | xargs -I {} echo "This is atom {}"'}]
+            },
+        'nt':
+            {
+                'commands': ['echo !TOKEN!> !ARTIFACT_DIR!\\result.txt'],
+                'atomizers': [{'TOKEN': 'FOR /l %n in (0,1,4) DO @echo This is atom %n'}]
+            },
```

Comment:
I'd like to keep these functional tests as yaml strings if possible -- I think it's a good thing that the test data is in the raw format. Maybe we should change the framework instead?